cv-data-service-library/src/main/java/com/trihydro/library/helpers/CdotGisConnector.java

```diff
@@ -0,0 +1,74 @@
+package com.trihydro.library.helpers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+
+import com.trihydro.library.service.RestTemplateProvider;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CdotGisConnector {
+  private final String baseUrl = "https://dtdapps.coloradodot.info/arcgis/rest/services/LRS/Routes/MapServer/exts/CdotLrsAccessRounded";
+
+  private RestTemplateProvider restTemplateProvider;
+
+  private final Logger logger = LoggerFactory.getLogger(CdotGisConnector.class);
+
+  @Autowired
+  public void InjectDependencies(RestTemplateProvider _restTemplateProvider) {
+    this.restTemplateProvider = _restTemplateProvider;
+  }
+
+  public String getBaseUrl() {
+    return baseUrl;
+  }
+
+  public RestTemplateProvider getRestTemplateProvider() {
+    return restTemplateProvider;
+  }
+
+  public ResponseEntity<String> getAllRoutes() {
+    String targetUrl = baseUrl + "/Routes";
+    logger.info("Getting all routes from CDOT GIS service at: {}", targetUrl);
+    String params = "?f=json";
+    HttpHeaders headers = new HttpHeaders();
+    headers.set("Accept", "application/json");
+    HttpEntity<String> entity = new HttpEntity<>(headers);
+    ResponseEntity<String> response;
+    try {
+      response = restTemplateProvider.GetRestTemplate().exchange(targetUrl + params, HttpMethod.GET, entity, String.class);
+    } catch (Exception e) {
+        logger.error("Error getting all routes from CDOT GIS service: {}", e.getMessage(), e);
+        return null;
+    }
+    if (response.getBody() != null) {
+      logger.info("Received response from CDOT GIS service: {}", response.getBody());
+    }
+    return response;
+  }
```

Comment:
suggestion: remove this unused code. We can easily rewrite this functionality should we find a need for it. This violates the YAGNI principle https://martinfowler.com/bliki/Yagni.html

Reply:
Thank you for this call-out; the article you linked was a great read. I'll remove this endpoint and the relevant service methods & tests.

Reply:
Removed unused `getAllRoutes()` method in CdotGisConnector & corresponding unit test!